src/sentry/integrations/utils/codecov.py

```diff
@@ -142,40 +142,41 @@ def get_codecov_data(repo: str, service: str, path: str) -> Tuple[LineCoverage |
     return line_coverage, codecov_url
 
 
-def fetch_codecov_data(config: Any) -> Tuple[Optional[Dict[str, Any]], Optional[str]]:
+def fetch_codecov_data(config: Dict[str, Any]) -> Dict[str, Any]:
+    data = {}
+    message = ""
     try:
         repo = config["repository"].name
         service = config["config"]["provider"]["key"]
         path = config["outcome"]["sourcePath"]
 
         lineCoverage, codecovUrl = get_codecov_data(repo, service, path)
         if lineCoverage and codecovUrl:
-            return {
+            data = {
                 "lineCoverage": lineCoverage,
                 "coverageUrl": codecovUrl,
                 "status": status.HTTP_200_OK,
-            }, None
+            }
     except requests.exceptions.HTTPError as error:
         data = {
             "attemptedUrl": error.response.url,
             "status": error.response.status_code,
         }
 
-        message = None
-        if error.response.status_code == status.HTTP_404_NOT_FOUND:
-            message = "Failed to get expected data from Codecov. Continuing execution."
```

Comment:
We will not be creating errors anymore when we get a 404 unless we really want to investigate them.

Here's a [query](https://sentry.sentry.io/discover/results/?field=title&field=count%28%29&field=codecov.http_code&field=codecov.timeout&name=Codecov+request+timed+out.+Continuing+execution.&project=1&query=issue%3ASENTRY-ZES&sort=-title&statsPeriod=12h&yAxis=count%28%29) of where the errors are being reported.